docs/rfc/rfc-2.adoc

```diff
@@ -0,0 +1,811 @@
+  index of the output pointing to the funding UTXO.
+- The previous output script, which is to say the address of the deposit.
+
+Notably, both of these pieces of data can be retrieved or computed solely
+from data available on the tBTC deposit contract (the previous outpoint
+cannot be retrieved solely from the ECDSA keep contract, but requires the
+associated tBTC deposit contract).
+
+Finally, once the per-signer amount and sighash are both known, constructing the
+signed Bitcoin transaction requires a few more pieces of information:
+
+- The signature of the sighash, which must be generated via the threshold ECDSA
+  functionality of the keep.
+- The public key of the current holder, which in this case is the keep and
+  should be computable locally by each signer.
+- For each of the signers, a Bitcoin address to receive the split amount.
+
+==== Required Inputs
+
+From this, we reach the full set of required inputs and where they can be
+procured or computed:
+
+- N, the number of signers in the group, available on the Keep contract
+  from the `getMembers` view method, but also tracked locally on the client
+  alongside the members of the keep.
+- L, the deposit lot size, available on the tBTC deposit contract from the
+  `lotSizeSatoshis` view method.
+- The previous outpoint describing the deposit UTXO, available on the tBTC
+  deposit contract from the `fundingInfo` view method, which returns it as
+  the third `utxoOutpoint` entry in its 3-part return tuple.
+- The keep's public key, available on the Keep contract from the
+  `getPublicKey` view method.
+- The deposit redemption script, straightforwardly computable from the keep's
+  public key by compressing the public key; this is briefly covered in the
+  implementation <<output-script>> section.
+- One Bitcoin address per signer, specified on the client in a way detailed
+  in the section on <<recovery-address>>.
+- The keep's signature on the sighash produced by the previous items,
+  computed via the threshold ECDSA protocol in a way detailed in the section on
+  the <<signing-protocol>>.
+- F, the desired fee, which is a function of the going feerate on the Bitcoin
+  network and the number of bytes in the final transaction.
+
+The desired fee is the most problematic one to procure under the goal of not
+strictly requiring a Bitcoin node connection, as generally a Bitcoin node
+provides estimates of the feerates that will achieve transaction confirmation
+within a certain amount of time. A short-term solution is proposed in the
+section on <<bitcoin-fee>>.
+
+[[recovery-address]]
+===== Specifying an Operator's Bitcoin Recovery Address
+
+To specify a given operator's Bitcoin Recovery Address, this proposal
+suggests adding a single key to the `Extensions.TBTC.BTCRefunds` config
+entry, `BeneficiaryAddress`. The value for this key should be one of:
+
+- A simple BTC address.
+- An xpub, ypub, or zpub that allows the client to derive new addresses for
```

Comment:
Why do we need to support both BTC address and *pubs?

Reply:
*pubs are best practice for multiple transactions in the Bitcoin world for privacy purposes, because they allow addresses that are publicly unrelated but privately controlled by the same key, making it less obvious who owns what. Some light explanation of hierarchical deterministic wallets, etc at https://coinguides.org/bitcoin-address-keep-changing/ .

Some folks don't care, and a simple BTC address shortcuts the address derivation process here.

Reply:
Okay - eventually the money needs to be sent to an address, and so the signer can either supply the address directly, or the signer can supply an address-generator (a *pub) to add a layer of abstraction, and we want to be able to support both in our first pass. :100: